src/main/scala/Tiebreaks.scala

```diff
@@ -0,0 +1,112 @@
+package strategygames.tiebreaks
+
+sealed trait Participant
+case class Player(id: String) extends Participant
+case object Virtual           extends Participant
+
+sealed trait Presence
+case object Absent  extends Presence
+case object Present extends Presence
+
+sealed trait POV
+case class Hero(participant: Player, presence: Presence) extends POV
+case class Foe(participant: Participant)                 extends POV
+
+// Results are always from a particualar point-of-view. So we only need
+// the three cases.
+sealed trait Outcome
+case object WonAgainst extends Outcome
+case object DrewWith   extends Outcome
+case object LostTo     extends Outcome
+
+case class Result(round: Int, player: Hero, outcome: Outcome, opponent: Foe)
+
+trait Tournament {
+  val nbRounds: Int
+  def resultsForPlayer(player: Player): List[Result]
+}
+
+class Tiebreak(val tournament: Tournament) {
+
+  val Win: Double  = 2
+  val Draw: Double = 1
+  val Loss: Double = 0
+  // NOTE: we use zero-base indexing here.
+
+  // In the following we distinguish between score and tiebreak.
+  // Score is what you get for the Result directly.
+  // Tiebreak is what you get from your opponent for a tiebreak
+  def scoreAtRound(p: Player, round: Int): Double =
+    tournament.resultsForPlayer(p).take(round).map(score).sum
+
+  def score(p: Player): Double =
+    scoreAtRound(p, tournament.nbRounds)
+
+  def virtualOpponentScoreAtRound(p: Player, round: Int): Double =
+    scoreAtRound(p, round) + remainingRoundDraws(round)
```

Comment:
I thought this should read as score from (round -1) but given round is 0 index based it works.....
However the funciton for remainingRoundDraws should not have the -1 (e.g. your mini tournament with 3 rounds and bye in final round for lakin2 -> gives (3 - 2 - 1)*1 = 0 but should give (3-2)*1 = 1?